bibliopixel/led.py

```diff
@@ -18,9 +18,9 @@ def __init__(self, driver):
         self._data = []
         self._driver = driver
 
-    def setData(self, data):
+    def setData(self, colors, pos):
         self._reading.wait()
-        self._data = data
+        self._data = (colors, pos)
```

Comment:
A variable that clearly needs more documentation!  :-D

You have a pessimization where for each driver you create a new sublist containing just the data for that driver.

So lots of memory management, lots of unnecessary copies.

Instead, I pass the root of the single list of pixels, and an offset from that last, and the drivers start reading from that offset within the list.

Copies eliminated!